index.d.ts

```diff
@@ -0,0 +1,677 @@
+   */
+  maxLength?: number;
+  minLength?: number;
+  pattern?: string;
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.4
+   */
+  items?: JSONSchema7Definition | JSONSchema7Definition[];
+  additionalItems?: JSONSchema7Definition;
+  maxItems?: number;
+  minItems?: number;
+  uniqueItems?: boolean;
+  contains?: JSONSchema7;
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.5
+   */
+  maxProperties?: number;
+  minProperties?: number;
+  required?: string[];
+  properties?: {
+    [key: string]: JSONSchema7Definition;
+  };
+  patternProperties?: {
+    [key: string]: JSONSchema7Definition;
+  };
+  additionalProperties?: JSONSchema7Definition;
+  dependencies?: {
+    [key: string]: JSONSchema7Definition | string[];
+  };
+  propertyNames?: JSONSchema7Definition;
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.6
+   */
+  if?: JSONSchema7Definition;
+  then?: JSONSchema7Definition;
+  else?: JSONSchema7Definition;
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.7
+   */
+  allOf?: JSONSchema7Definition[];
+  anyOf?: JSONSchema7Definition[];
+  oneOf?: JSONSchema7Definition[];
+  not?: JSONSchema7Definition;
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-7
+   */
+  format?: string;
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-8
+   */
+  contentMediaType?: string;
+  contentEncoding?: string;
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-9
+   */
+  definitions?: {
+    [key: string]: JSONSchema7Definition;
+  };
+
+  /**
+   * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-10
+   */
+  title?: string;
+  description?: string;
+  default?: JSONSchema7Type;
+  readOnly?: boolean;
+  writeOnly?: boolean;
+  examples?: JSONSchema7Type;
+}
+
+import {ErrorObject} from 'ajv';
+
+type Schema = JSONSchema4 | JSONSchema6 | JSONSchema7;
+type PostFormatter = (formattedError: string, error: ErrorObject) => string;
+
+declare namespace SchemaUtils {
+  class ValidateError extends Error {
+    constructor(errors: Array<ErrorObject>, schema: Schema, configuration?: Partial<ValidateErrorConfiguration>);
+
+    name: string;
+    errors: Array<ErrorObject>;
+    schema: Schema;
+    headerName: string;
+    baseDataPath: string;
+    postFormatter: PostFormatter | null;
+    message: string;
+  }
+
+  interface ValidateErrorConfiguration {
+    name: string,
+    baseDataPath: string,
+    postFormatter: PostFormatter
+  }
+}
+
+declare var validate: {
+  (schema: Schema, options: Array<object> | object, configuration?: Partial<SchemaUtils.ValidateErrorConfiguration>): void;
+  ValidateError: typeof SchemaUtils.ValidateError
+}
+
+export = validate;
```

Comment:
Why don't add some types in devDeps?

Reply:
which one?

Reply:
`@types/json-schema`

Reply:
Hmmm, I just followed your previous suggestions -( Ok i will revert this and move this to dev dependency

Reply:
@vankop Sometimes I have a lot issues and lose context, sorry for that, all make mistakes :smile:

Reply:
It is fine